src/cmds/restic/cmd_forget.go

```diff
@@ -114,37 +79,33 @@ func runForget(opts ForgetOptions, gopts GlobalOptions, args []string) error {
 		return err
 	}
 
-	// parse arguments as hex strings
-	var ids []string
-	for _, s := range args {
-		_, err := hex.DecodeString(s)
-		if err != nil {
-			Warnf("argument %q is not a snapshot ID, ignoring\n", s)
-			continue
-		}
-
-		ids = append(ids, s)
-	}
-
-	// process all snapshot IDs given as arguments
-	for _, s := range ids {
-		id, err := restic.FindSnapshot(repo, s)
-		if err != nil {
-			Warnf("could not find a snapshot for ID %q, ignoring\n", s)
-			continue
-		}
-
-		if !opts.DryRun {
-			h := restic.Handle{Type: restic.SnapshotFile, Name: id.String()}
-			err = repo.Backend().Remove(h)
+	// Process all snapshot IDs given as arguments.
+	if len(args) != 0 {
+		for _, s := range args {
```

Comment:
No, just make a new function in `cmd_forget.go`, so that this function becomes shorter.